src/components/CodeView/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+
+import styles from './styles.module.scss';
+import { getLanguage, getLines, mapWithDepth } from './utils';
+import refractor from '../../refractor';
+
+type PublicProps = {
+  mimeType: string;
+  content: string;
+};
+
+// This function mimics what https://github.com/rexxars/react-refractor does,
+// but we need a different layout to inline comments so we cannot use this
+// component.
+const renderHighlightedCode = (code: string, language: string) => {
+  const ast = refractor.highlight(code, language);
+  const value = ast.length === 0 ? code : ast.map(mapWithDepth(0));
+
+  return (
+    <pre className={styles.highlightedCode}>
+      <code className={`language-${language}`}>{value}</code>
```

Comment:
`value` doesn't seem to be tested at all. All the tests pass for me with this:

```diff
diff --git a/src/components/CodeView/index.tsx b/src/components/CodeView/index.tsx
index 1b4bbad..cbf0d90 100644
--- a/src/components/CodeView/index.tsx
+++ b/src/components/CodeView/index.tsx
@@ -18,7 +18,7 @@ const renderHighlightedCode = (code: string, language: string) => {
 
   return (
     <pre className={styles.highlightedCode}>
-      <code className={`language-${language}`}>{value}</code>
+      <code className={`language-${language}`}></code>
     </pre>
   );
 };
```

Can you add tests for both of `ast` cases?

Reply:
yep, thanks